src/main/java/uk/gov/hmcts/darts/common/service/impl/CaseCommonServiceImpl.java

```diff
@@ -0,0 +1,58 @@
+package uk.gov.hmcts.darts.common.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.hmcts.darts.common.entity.CourtCaseEntity;
+import uk.gov.hmcts.darts.common.entity.CourthouseEntity;
+import uk.gov.hmcts.darts.common.entity.UserAccountEntity;
+import uk.gov.hmcts.darts.common.repository.CaseRepository;
+import uk.gov.hmcts.darts.common.service.CaseCommonService;
+import uk.gov.hmcts.darts.common.service.CourthouseCommonService;
+
+import java.util.Optional;
+
+@RequiredArgsConstructor
+@Service
+public class CaseCommonServiceImpl implements CaseCommonService {
+
+    private final CaseRepository caseRepository;
+    private final CourthouseCommonService courthouseCommonService;
+
+    @Override
+    @Transactional
```

Comment:
If we are already inside a Transaction, one that was started outside of this class, will it create a new one, or continue to use the encapsulating one?
i.e. do we need propagation = Propagation.REQUIRES_NEW ?

Reply:
No we don't need propagation = Propagation.REQUIRES_NEW